posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+<tr>                                                       <td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td></tr>
+
+</table>
+
+Here are some trends I noted from looking at the table:
+
+The overwhelming majority of our ambitions, 19 out of 22, reported *some*
+amount of progress. That's great, especially given that several of those
+ambitions were mere aspirations that explicitly had no resources attached to
+them at the start of the year.
+
+Less than a quarter of the ambitions, 6 out of 22, said that their overall
+problem would be resolved, or the bulk of the problem addressed, within the next
+six months. Ten others said they predicted the important components of their
+solution would be available within the next two years, so that's a total of 16
+out of 22 that expect some kind of solution within two years.
+
+  * Note: One respondent noted that the two year threshold embedded in the
+    survey was an "odd timeframe." I do not disagree with that remark. The
+    reality is that I was trying to find a happy medium between two extremes:
+    short timeframes don't allow enough time for significant work to be
+    accomplished, while excessively long timeframes (e.g. greater than five
+    years) are very difficult to predict realistically. But the two year
+    time frame was drawn somewhat out of a hat.
+
+Of the nineteen ambitions that reported some progress so far this year, fourteen
+of them (14/19) also said that at the start of the year, they had either *no
+goals* or *no plan* to achieve those goals. Ten of those fourteen say they now
```

Comment:
I think this framing is not quite accurate. From what I can see, every initiative/aspiration has some kind of goal (stabilize feature x, improve problem y, etc), it was having a concrete plan to achieve the goal that was lacking. Many of the ambitions are open ended as well and another interpretation of the data for those groups is that "there is no plan that will achieve the goal because it is never ending" (ie, we can always make diagnostics/debuginfo/compiler perf/etc better).

Reply:
I had a bunch more private conversation with Wesley about this and I think we're okay with the text after the subsequent revisions we made on 2022-08-05.